frontend/src/api/k8s/__tests__/notebooks.spec.ts

```diff
@@ -0,0 +1,936 @@
+    expect(renderResult).toStrictEqual(mockNotebookK8sResource({ uid }));
+  });
+  it('should create a notebook with pipelines', async () => {
+    const Notebook = assembleNotebook(mockStartNotebookData({}), username, true);
+    k8sCreateResourceMock.mockResolvedValue(mockNotebookK8sResource({ uid }));
+
+    createElyraServiceAccountRoleBindingMock.mockResolvedValue(mockRoleBindingK8sResource({ uid }));
+
+    const renderResult = await createNotebook(mockStartNotebookData({}), username, true);
+
+    expect(k8sCreateResourceMock).toHaveBeenCalledWith({
+      model: NotebookModel,
+      resource: Notebook,
+    });
+    expect(createElyraServiceAccountRoleBindingMock).toHaveBeenCalledTimes(1);
+    expect(k8sCreateResourceMock).toHaveBeenCalledTimes(1);
+    expect(renderResult).toStrictEqual(mockNotebookK8sResource({ uid }));
+  });
+  it('should handle errors and rethrow', async () => {
+    const Notebook = assembleNotebook(mockStartNotebookData({}), username, true);
+    k8sCreateResourceMock.mockRejectedValue(new Error('error1'));
+    await expect(createNotebook(mockStartNotebookData({}), username, true)).rejects.toThrow(
+      'error1',
+    );
+    expect(k8sCreateResourceMock).toHaveBeenCalledTimes(1);
+    expect(k8sCreateResourceMock).toHaveBeenCalledWith({
+      model: NotebookModel,
+      resource: Notebook,
+    });
+  });
+});
+
+describe('getNotebook', () => {
+  it('should get a notebook', async () => {
+    const name = 'test-notebook';
+    const namespace = 'test-project';
+
+    k8sGetResourceMock.mockResolvedValue(mockNotebookK8sResource({ uid }));
+
+    const renderResult = await getNotebook(name, namespace);
+
+    expect(k8sGetResourceMock).toHaveBeenCalledWith({
+      model: NotebookModel,
+      queryOptions: { name, ns: namespace },
+    });
+    expect(k8sGetResourceMock).toHaveBeenCalledTimes(1);
+    expect(renderResult).toStrictEqual(mockNotebookK8sResource({ uid }));
+  });
+  it('should handle errors and rethrow', async () => {
+    const name = 'test-notebook';
+    const namespace = 'test-project';
+
+    k8sGetResourceMock.mockRejectedValue(new Error('error1'));
+    await expect(getNotebook(name, namespace)).rejects.toThrow('error1');
+    expect(k8sGetResourceMock).toHaveBeenCalledTimes(1);
+    expect(k8sGetResourceMock).toHaveBeenCalledWith({
+      model: NotebookModel,
+      queryOptions: { name, ns: namespace },
+    });
+  });
+});
+describe('getNotebooks', () => {
+  it('should get all notebooks', async () => {
+    const namespace = 'test-project';
+
+    k8sListResourceMock.mockResolvedValue(mockK8sResourceList([mockNotebookK8sResource({ uid })]));
+
+    const renderResult = await getNotebooks(namespace);
+    expect(k8sListResourceMock).toHaveBeenCalledWith({
+      model: NotebookModel,
+      queryOptions: { ns: namespace },
+    });
+    expect(k8sListResourceMock).toHaveBeenCalledTimes(1);
+    expect(renderResult).toStrictEqual([mockNotebookK8sResource({ uid })]);
```

Comment:
Create a mock instance once in this function and reuse it if the expectation is that these two instances are the same values.